packages/agent/src/agent/http/index.ts

```diff
@@ -532,7 +532,22 @@ export class HttpAgent implements Agent {
     if (!this.#verifyQuerySignatures) {
       return query;
     }
-    return this.#verifyQueryResponse(query, subnetStatus);
+    try {
+      return this.#verifyQueryResponse(query, subnetStatus);
+    } catch (_) {
+      // In case the node signatures have changed, refresh the subnet keys and try again
+      console.warn('Query response verification failed. Retrying with fresh subnet keys.');
+      this.#subnetKeys.delete(canisterId.toString());
+      await this.fetchSubnetKeys(canisterId.toString());
+
+      const updatedSubnetStatus = this.#subnetKeys.get(canisterId.toString());
+      if (!updatedSubnetStatus) {
+        throw new CertificateVerificationError(
+          'Invalid signature from replica signed query: no matching node key found.',
+        );
+      }
+      return this.#verifyQueryResponse(query, updatedSubnetStatus);
```

Comment:
Is it okay that there is no `resolve(subnetStatus);` call here, in between `get` and `verifyQueryResponse`, as in the above code?

Reply:
Yes, we are not within a `Promise` at this stage in the code (apart from the `async query` function itself). It's a little semantically incongruous if you're comparing this section to the above sections, however